### ISSUE TITLE
The order of remote and local is inconsistent

### ISSUE THREAD

**Issue by huang12zheng** [flutterdata/flutter_data#167]
# What the problem looks like
Confirmed that the problem ([bad data](https://github.com/flutterdata/flutter_data/issues/166#issuecomment-1140966789)) is due to remote data
Or maybe it's the Internet.
10 clicks in a row, local 10 is correct.
After the local end, the remote data will be returned.
But it is disorder.

* _RemoteAdapter.save
```dart
var rng = math.Random().nextInt(100000); ///<<<<
print(rng); ///<<<<
final serialized = await serialize(model);
...
final result = await sendRequest<T>(
...
  onSuccess: (data, label) {
    onSuccess ?? = (data, label, _) => this.onSuccess<T>(data, label);
    print(rng); ///<<<<
```
---
localstate:
55436 41050 82477 22148 40731 22279 43336
remotestate:
40731 55436 22279 41050 43336 82477 22148

# What I want
* solution1
One possibility is that I can `cancel` the remote request and be sure to save the last local request, which is 43336
> it maybe look like [Riverpod Example: Canceling HTTP requests when no longer used](https://riverpod.dev/docs/concepts/modifiers/auto_dispose/#example-canceling-http-requests-when-no-longer-used)
But it requires too many users to control it themselves
* solution2
reorder remote reply.
we can save data 22148 true, behind 82477 false,But we could change UI state.
Due to that I have received 40731.

**Comment by frank06** [comment]
Okay I get what you mean and I think I fixed it here: 33f18fd22add38cef70bf0129d5facba26ce92ea

I also updated https://github.com/flutterdata/flutter_data_todos with it, can you try if it works better now?

**Comment by huang12zheng** [comment]
It looks good,
Specifically,
Data that does not need to be updated is not updated.
